### PR TITLE
JAMES-3028 Split blobstore memory

### PR DESCRIPTION
**Pull request by mbaechler**
Rebase of https://github.com/linagora/james-project/pull/3045

### CODE REVIEW

File: server/blob/blob-api/src/main/java/org/apache/james/blob/api/DumbBlobStore.java [hunk 1]
```diff
@@ -51,6 +51,7 @@
      * Save the blob with the provided blob id, and overwrite the previous blob with the same id if it already exists
      * The bucket is created if it not already exists.
      * This operation should be atomic and isolated
+     * Two blobs having the same blobId must have the same content
```
Comment (rouazana) [inline server/blob/blob-api/src/main/java/org/apache/james/blob/api/DumbBlobStore.java]: note for later: we have a little issues with our comments:
- some are for people who will use the API
- some are for people who will implement the API
We should clearly differentiate between both

File: server/blob/blob-api/src/test/java/org/apache/james/blob/api/DumbBlobStoreFixture.java [hunk 2]
```diff
@@ -31,7 +31,7 @@
     byte[] EMPTY_BYTEARRAY = {};
     byte[] SHORT_BYTEARRAY = SHORT_STRING.getBytes(StandardCharsets.UTF_8);
     byte[] ELEVEN_KILOBYTES = Strings.repeat("2103456789\n", 1000).getBytes(StandardCharsets.UTF_8);
-    byte[] TWELVE_MEGABYTES = Strings.repeat("0123456789\r\n", 1024 * 10).getBytes(StandardCharsets.UTF_8);
     String TWELVE_MEGABYTES_STRING = Strings.repeat("7893456789\r\n", 1024 * 10);
```
Comment (rouazana) [inline server/blob/blob-api/src/test/java/org/apache/james/blob/api/DumbBlobStoreFixture.java]: that's not twelve MB!

Comment (mbaechler) [reply server/blob/blob-api/src/test/java/org/apache/james/blob/api/DumbBlobStoreFixture.java]: fixed

**Comment by mbaechler** [comment]
merged

**Pull request closed without merging**